### DISCUSSION TITLE
Python access to chunk options

### DISCUSSION THREAD

**Discussion by jdnewmil** [quarto-dev/quarto-cli#2561] [Q&A]
Is it possible to refer to chunk options in a python chunk?

Hypothetical use case would be for managing fig-width and fig-height on a per-chunk basis in a manner consistent with the yml header. See hypothetical code below:

```{python}
#! fig-width: 8
#! fig-height: 10
import plotnine as p9
import plotnine.data as p9d
print(
  p9.ggplot(
    p9d.mpg
    , p9.aes(x='disp', y='mpg'))
  + p9.geom_points()
  + p9.theme(figure_size=(chunk_opt['fig-width'], chunk_opt['fig-height']))
```

**Comment by mcanouil** [comment]
From the reference guide, those options aren't valid options for Jupyter engine (<https://quarto.org/docs/reference/cells/cells-jupyter.html>).

Currently, the options are only used for matplotlib rcParams `figure.figsize`.

I do not think there is a way to read from code cell options before evaluating the code cell (not even sure a Lua filter will intervene at the proper time to deal with that).

**Comment by jdnewmil** [comment]
I know that, but setting the figure size per chunk is straightforward with the knitr engine so my question is about accessing chunk options inside the chunk to emulate that ability. The code is only intended to convey the desired linkage... please focus on the question if you cannot follow the gist of the code.

On September 23, 2022 6:41:03 PM PDT, "Mickaël Canouil" ***@***.***> wrote:
>From the reference guide, those options aren't valid options for Jupyter engine (<https://quarto.org/docs/reference/cells/cells-jupyter.html>).
>

-- 
Sent from my phone. Please excuse my brevity.

**Reply by mcanouil** [reply]
> please focus on the question if you cannot follow the gist of the code.

You could also read the complete answer and not only the first part (which you seem to only have because of email notification).

In addition, from the code base, for your example about figure size, it is set by a Python script ran before any computation (<https://github.com/mcanouil/quarto-cli/blob/main/src/resources/jupyter/lang/python/setup.py#L15-L23> and <https://github.com/mcanouil/quarto-cli/blob/main/src/resources/jupyter/notebook.py#L69-L80>).
Maybe using a pre-render script, you could achieve what you want (<https://quarto.org/docs/projects/scripts.html#pre-and-post-render>).
